alibi_detect/saving/_tensorflow/loading.py

```diff
@@ -36,36 +37,31 @@
 
 
 def load_model(filepath: Union[str, os.PathLike],
-               filename: str = 'model',
-               custom_objects: dict = None,
                layer: Optional[int] = None,
```

Comment:
Minor, but I don't fully agree with omitting `filename` from these internal save/load functions. What we save for in function signature, we pay at every callsite, having to remember to do `.joinpath(filename)`. (OTOH in the old behaviour, having a default `model` name is also likely not desirable as forgetting to set it would result in a perhaps unexpected default).

Reply:
Yeh very true. Pretty much the only reason I made this change is that the `filename` is only there for legacy loading (legacy as in, saving to `.h5`, and legacy as in loading files with different names such as `encoder.h5`). For the "modern" loading we simply do:

```python
    if flavour == Framework.TENSORFLOW:
        model = load_model_tf(src, layer=layer, **kwargs)
```

So I saw it as a trade-off wrt to carrying around more complexity in `load_model` to facilitate the legacy functionality in `load_detector_legacy`, or add some complexity to the calls in `load_detector_legacy` to simplify the `load_model` function... (which just happens to be used for modern and legacy loading). 

Similar story for `save_model`...

Reply:
> Also, noting that there's a few subtle changes in the behaviour of internal functions for saving/loading, need to be extra vigilant new bugs haven't been introduced.

p.s. below is very true though... tweaking anything to do with legacy save/load does bring the potential for bugs like the ones `v0.10.5` fixed. I've run the same tests of loading old artefacts we ran in https://github.com/SeldonIO/alibi-detect/pull/732 and everything passes, but that isn't 100% comprehensive...